src/main/java/com/google/sps/servlets/Utils.java

```diff
@@ -86,6 +90,36 @@ static Calendar loadCalendarClient() throws IOException {
     return new Calendar.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential).build();
   }
 
+  // Query datastore with the current users email to retrieve user info entity
+  static Entity getCurrentUserEntity() {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    return results.asSingleEntity();
+  }
+
+  // Retrieve the family id from the user info and fetch their family entity from datastore
+  static Entity getCurrentFamilyEntity(Entity userInfoEntity) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    long familyID = (long) userInfoEntity.getProperty("familyID");
+
+    Key familyEntityKey = KeyFactory.createKey("Family", familyID);
+
+    Entity familyEntity = null;
+    try {
+        familyEntity = datastore.get(familyEntityKey);
+    } catch (EntityNotFoundException e) {
+        System.out.println("Family not found");
```

Comment:
If this is unexpected (the family should never be null), consider throwing the exception (as a checked exception), and expect the caller to handle it.

Otherwise, make sure you're checking for nullness in all the calls.

You can also consider using java.util.Optional, which makes it easier to avoid null pointer exceptions.

Reply:
Yea the family should never be null, changed to throw checked exception